contracts/misc/EthRewardSplitter.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity ^0.8.22;
+
+import {Address} from '@openzeppelin/contracts/utils/Address.sol';
+import {ReentrancyGuard} from '@openzeppelin/contracts/utils/ReentrancyGuard.sol';
```

Comment:
You must use upgradeable version of the ReentrancyGuard as it has an initializer that must be called.